lib/beacon_api/controllers/v1/beacon_controller.ex

```diff
@@ -6,12 +6,11 @@ defmodule BeaconApi.V1.BeaconController do
 
   @spec get_state_root(Plug.Conn.t(), any) :: Plug.Conn.t()
   def get_state_root(conn, %{"state_id" => state_id}) do
-    case BeaconApi.Utils.parse_id(state_id) |> ForkChoice.Helpers.root_by_id() do
-      {:ok, {root, execution_optimistic, finalized}} ->
-        with {:ok, state_root} <- ForkChoice.Helpers.get_state_root(root) do
-          conn |> root_response(state_root, execution_optimistic, finalized)
-        end
-
+    with {:ok, {root, execution_optimistic, finalized}} <-
+           BeaconApi.Utils.parse_id(state_id) |> ForkChoice.Helpers.root_by_id(),
+         {:ok, state_root} <- ForkChoice.Helpers.get_state_root(root) do
```

Comment:
we should try to avoid getters to ForkChoice, instead using BeaconChain

Reply:
This doesn't access the `ForkChoice` `GenServer`. It just uses some helper methods, although we probably should change the module's name 😅